EventsExpress/ClientApp/src/components/location/location.js

```diff
@@ -0,0 +1,99 @@
+import React, { Component } from "react";
+import "react-widgets/dist/css/react-widgets.css";
+import { LocationMapWithMarker } from "../helpers/form-helpers";
+import { enumLocationType } from "../../constants/EventLocationType";
+import FormControlLabel from "@material-ui/core/FormControlLabel";
+import Radio from "@material-ui/core/Radio";
+import { renderFieldError } from "../helpers/form-helpers";
+import FormControl from "@material-ui/core/FormControl";
+import RadioGroup from "@material-ui/core/RadioGroup";
+import TextField from "@material-ui/core/TextField";
+export default class Location extends Component {
+  onChangeLocationType = (event) => {
+    const type = Number(event.target.value);
+    if (type === enumLocationType.map) {
+      this.props.input.onChange({ type, latitude: null, longitude: null });
+    } else if (type === enumLocationType.online) {
+      this.props.input.onChange({ type, onlineMeeting: null });
+    }
+  };
+
+  onUrlInputChange = (event) => {
+    this.props.input.onChange({
+      type: enumLocationType.online,
+      onlineMeeting: event.target.value === "" ? null : event.target.value,
+    });
+  };
+
+  onMapLocationChange = (mapLocation) =>{
+    this.props.input.onChange({
+      type:enumLocationType.map,
+      latitude:mapLocation.latitude,
+      longitude:mapLocation.longitude
+    })
+  }
+
+  returnLocationRender = () => {
```

Comment:
this should be simplified ad merged with the `render()` method

Reply:
i need this function because if i render it directly on render() function some error occurs, i'll explin at daily meeting this

Reply:
resolved